src/boot/memory.c

```diff
@@ -386,6 +386,11 @@ void *load_segment_decompress(s32 segment, u8 *srcStart, u8 *srcEnd) {
     void *dest = NULL;
 
     u32 compSize = ALIGN16(srcEnd - srcStart);
+
+#ifdef GZIP
+    struct libdeflate_decompressor *dec = libdeflate_alloc_decompressor();
```

Comment:
Can this be given a NULL check and aggress if it fails? The game should probably just cleanly die whenever this happens.